bin/image

```diff
@@ -28,6 +28,16 @@ last4minors() {
     done
 }
 
+# filters kube versions which are already pushed
+# missing_from_repo <VERSION...>
+missing_from_repo() {
+    present_tags=$(curl -sSL https://registry.hub.docker.com/v1/repositories/claranet/gcloud-kubectl-docker/tags)
```

Comment:
please do this once, so you can avoid fetching external resources for each call to `missing_from_repo`. Another function might be a good solution here, like `fetch_dockerhub_tags()`

Reply:
This is only called once since it looks up all versions in single function call, but the script is run for each operation (build, test, push) separately. So this function is usually called 2-3 times over the entire job. If I want to keep the operations independent from one another I would have to save the versions externally.

Reply:
you're right, sorry; didn't catch that